### ISSUE TITLE
Grbl Error 25

### ISSUE THREAD

**Issue by mompes** [synman/Octoprint-Bettergrblsupport#3]
I have just installed Octoprint and Better grbl support on a raspberry pi and I am getting the error 25 (Invalid gcode ID:25 Repeated g-code word found in block.). I have tried a couple of times and it doesn't happens at the same moment, but it happens very quickly, just 10-15 seconds after starting.
Do you have any advice about it?

**Comment by synman** [comment]
What program are you using as your slicer?   If you enable logging within Octoprint you should be able to get a "dump" of the actual gcode being streamed.  I'm guessing either my parser is choking on an unrecognized set of commands or your engraver doesn't like it.

What engraver are you using?   2 or 3 axes?   More than happy to help troubleshoot... would be awesome to see your octoprint log with debugging enabled when it chokes.

**Comment by thothloki** [comment]
I am trying to help him as well. I have octopi loaded with your plugin, i just need to test it.
I am using T2Laser as my slicer and will be using an EleksMaker Mana SE and a L7 board as my controller
2 axis A3 machine

What slicer do you use? @synman

**Comment by synman** [comment]
Lightburn

**Comment by thothloki** [comment]
> Lightburn

are you able to post an example gcode file? mgiht be something useful to be able to test if it is our slicer or hardware. Especially if it is a known working gcode file.

**Comment by synman** [comment]
Provide your comm logs and Octoprint logs, ideally in debug or verbose format, if you want me to look into it.

Alternatively, you are more than welcome to fork the code, troubleshoot it yourself, and submit a pull request if you find something others can benefit from.

**Comment by thothloki** [comment]
Will do. I am trying to help Mompes on this issue and was curious to compare the lightburn gcode to T2Laser gcode to see if there is much of a difference.
I have not been able to hook this up to one of my engravers yet. Life keeps getting in the way

**Comment by thothloki** [comment]
@synman i will most likely have to fork this. It does not take .nc files (the file format that T2Laser generates) and i got an error pretty quick (22 i think). The only think I am thinking of changing is the allowed file formats and the gcode sending code.
I think you set yours up to specifically work with lightburn

**Comment by thothloki** [comment]
I think I found the issue that I am having. Lightburn gcode is not optimized as T2Laser's is and you use the "G/M" codes at the beginning of each line to process your gcode. 
I was able to export that kind of gcode file, but it still screwed up the size of my engraving. it said a 175x151mm engraving was somewhere over 8 million mm each direction.
This is an awesome plugin, but I suggest you add a notice on your plugin that it is optimized for Lightburn and will require unoptimized gcode
